scripts/release.ts

```diff
@@ -0,0 +1,265 @@
+  const { version } = JSON.parse(
+    readFileSync(join(packages["web-features"], "package.json"), {
+      encoding: "utf-8",
+    })
+  );
+
+  // Commit
+  logger.info("Committing version bump");
+  const commitMessage = `Increment ${args.semverlevel} version to v${version}`;
+  const commitCmd = `git commit --all --message="${commitMessage}"`;
+  run(commitCmd);
+
+  // Push release branch
+  logger.info("Pushing release branch");
+  const pushCmd = `git push --set-upstream origin ${releaseBranch}`;
+  run(pushCmd);
+
+  // Create PR
+  logger.info(`Creating PR for ${version}`);
+  const title = `📦 Release web-features@${version}`;
+  const reviewer = "ddbeck";
+  const bodyFile = fileURLToPath(
+    new URL("release-pull-description.md", import.meta.url)
+  );
+
+  const temporaryDir = mkdtempSync(join(tmpdir(), "pull-request-"));
+  const temporaryBodyFile = join(temporaryDir, "body.md");
+  copyFileSync(bodyFile, temporaryBodyFile);
+  appendFileSync(temporaryBodyFile, ["```diff", diff, "```"].join("\n"));
+
+  const pullRequestCmd = [
+    "gh pr create",
+    `--title="${title}"`,
+    `--reviewer="${reviewer}"`,
+    `--body-file=${temporaryBodyFile}`,
+    `--base="main"`,
+    `--head="${releaseBranch}"`,
+    // `--repo="ddbeck/feature-set"`, // Uncomment this line to point this at a fork for testing
+  ].join(" ");
+  run(pullRequestCmd);
+
+  rmSync(temporaryDir, { recursive: true });
+}
+
+function update(args) {
+  // TODO: Rebase
+  // TODO: Run `npm run build`
+  // TODO: Commit results of `npm run build`
+  // TODO: Update PR diff
+  throw Error("Not implemented");
+}
+
+function publish(args) {
+  // TODO: Run `npm publish …`
+  throw Error("Not implemented");
+}
+
+function run(cmd: string) {
+  execSync(cmd, { stdio: "inherit" });
+}
+
+function diffJson(): string {
+  const temporaryDir = mkdtempSync(join(tmpdir(), "web-features-"));
+
+  execSync("npm install web-features", {
+    cwd: temporaryDir,
+    encoding: "utf-8",
+  });
+
+  const releasedJson = join(
+    temporaryDir,
+    "node_modules",
+    "web-features",
+    "index.json"
+  );
+  const prettyReleasedJson = execSync(`jq . "${releasedJson}"`, {
+    encoding: "utf-8",
+  });
+  const prettyReleasedJsonFp = join(temporaryDir, "index.released.pretty.json");
+  writeFileSync(prettyReleasedJsonFp, prettyReleasedJson);
+
+  execSync("npm run build", { stdio: "inherit" });
+  const preparedJson = join(packages["web-features"], "index.json");
+  const prettyPreparedJson = execSync(`jq . "${preparedJson}"`, {
+    encoding: "utf-8",
+  });
+  const prettyPreparedJsonFp = join(temporaryDir, "index.prepared.pretty.json");
+  writeFileSync(prettyPreparedJsonFp, prettyPreparedJson);
+
+  try {
+    const result = execSync(
+      `diff --unified "${prettyReleasedJsonFp}" "${prettyPreparedJsonFp}"`,
+      { encoding: "utf-8" }
+    );
+    rmSync(temporaryDir, { recursive: true });
+    return result;
+  } catch (err) {
+    if (err.status === 1) {
```

Comment:
Can you comment what this all is about?

Reply:
Good idea. Added a comment with 36c9de6.